platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/chassis.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+#############################################################################
+# DELLEMC
+#
+# Module contains an implementation of SONiC Platform Base API and
+# provides the platform information
+#
+#############################################################################
+
+try:
+    import os
+    from sonic_platform_base.chassis_base import ChassisBase
+except ImportError as e:
+    raise ImportError(str(e) + "- required module not found")
+
+
+class Chassis(ChassisBase):
+    """
+    DELLEMC Platform-specific Chassis class
+    """
+
+    HWMON_DIR = "/sys/devices/platform/SMF.512/hwmon/"
+    HWMON_NODE = os.listdir(HWMON_DIR)[0]
+    MAILBOX_DIR = HWMON_DIR + HWMON_NODE
+
+    reset_reason_dict = {}
+    reset_reason_dict[11] = ChassisBase.REBOOT_CAUSE_POWER_LOSS
+    reset_reason_dict[33] = ChassisBase.REBOOT_CAUSE_WATCHDOG
+    reset_reason_dict[44] = ChassisBase.REBOOT_CAUSE_NON_HARDWARE
+    reset_reason_dict[55] = ChassisBase.REBOOT_CAUSE_NON_HARDWARE
+
+    power_reason_dict = {}
+    power_reason_dict[11] = ChassisBase.REBOOT_CAUSE_POWER_LOSS
+    power_reason_dict[22] = ChassisBase.REBOOT_CAUSE_THERMAL_OVERLOAD_CPU
+    power_reason_dict[33] = ChassisBase.REBOOT_CAUSE_THERMAL_OVERLOAD_ASIC
+    power_reason_dict[44] = ChassisBase.REBOOT_CAUSE_INSUFFICIENT_FAN
```

Comment:
FYI, yesterday I changed this constant name to `REBOOT_CAUSE_INSUFFICIENT_FAN_SPEED`. Please update accordingly.

Reply:
Hi @jleveque 
Updated the constant name.